src/riak_kv_qry_compiler.erl

```diff
@@ -58,28 +57,43 @@
               sorter/0]).
 -export_type([where_props/0]).
 
--define(MAX_QUERY_QUANTA, 1000).  %% cap the number of subqueries the compiler will emit
+-define(MAX_QUERY_QUANTA, 1000).
+%% cap the number of subqueries the compiler will emit
 %% Note that when such a query begins to be actually executed, the
 %% chunks will need to be really small, for the result to be within
 %% max query size.
 
+compile(DDL, Query) ->
+    compile(DDL, Query, options()).
 
--spec compile(?DDL{}, ?SQL_SELECT{}) ->
+-spec compile(?DDL{}, ?SQL_SELECT{}, Options::options()) ->
     {ok, [?SQL_SELECT{}]} | {error, any()}.
-compile(?DDL{}, ?SQL_SELECT{is_executable = true}) ->
+compile(?DDL{}, ?SQL_SELECT{is_executable = true}, _) ->
     {error, 'query is already compiled'};
-compile(?DDL{table = T} = DDL,
-        ?SQL_SELECT{is_executable = false} = Q1) ->
+compile(?DDL{table = T} = DDL, ?SQL_SELECT{is_executable = false} = Q1, Options) ->
     Mod = riak_ql_ddl:make_module_name(T),
     case compile_order_by(
            maybe_compile_group_by(
-             Mod, compile_select_clause(DDL, Q1), Q1)) of
+             Mod, compile_select_clause(DDL, Options, Q1), Q1)) of
         {ok, Q2} ->
-            compile_where_clause(DDL, Q2);
+            compile_where_clause(DDL, Q2, Options);
         {error, _} = Error ->
             Error
     end.
 
+%% create a proplist of options for query compilation e.g. configured or
+%% transient values.
+options() ->
```

Comment:
It might be large but it is single responsibility, a compiler with options is understandable I think. For example the `erl_parse` module just calls `Options`, `Options`. 

http://erlang.org/doc/man/erl_parse.html#abstract-2